src/wobbly/handlers/service.py

```diff
@@ -0,0 +1,174 @@
+"""Routes used by other services.
+
+These routes should be protected by a service-only ``GafaelfawrIngress`` and
+thus only be accessible to other services using delegated credentials. The
+service and username can then be extracted from the header information added
+by Gafaelfawr and used to limit the view of the UWS database accessible to the
+service.
+"""
+
+from datetime import datetime
+from typing import Annotated, Any
+
+from fastapi import APIRouter, Depends, Header, Path, Query, Response
+from safir.dependencies.gafaelfawr import auth_dependency
+from safir.models import ErrorLocation
+from vo_models.uws.types import ExecutionPhase
+
+from ..dependencies.context import RequestContext, context_dependency
+from ..exceptions import UnknownJobError
+from ..models import Job, JobCreate, JobIdentifier, JobUpdate
+
+__all__ = ["router"]
+
+router = APIRouter()
+"""FastAPI router for all internal handlers."""
+
+
+async def auth_service_dependency(
+    x_auth_request_service: Annotated[str, Header(include_in_schema=False)],
+) -> str:
+    """Get the authenticated service from the Gaelfawr headers.
+
+    Returns
+    -------
+    str
+        Identifier of the service making the request.
+    """
+    return x_auth_request_service
+
+
+async def job_identifier_dependency(
+    *,
+    service: Annotated[str, Depends(auth_service_dependency)],
+    user: Annotated[str, Depends(auth_dependency)],
+    job_id: Annotated[str, Path(title="Job ID")],
+) -> JobIdentifier:
+    return JobIdentifier(service=service, owner=user, id=job_id)
+
+
+@router.get(
+    "/jobs",
+    description="List the jobs for the authenticated user",
+    response_model=list[Job],
+    response_model_exclude_defaults=True,
```

Comment:
Why `response_model_exclude_defaults` on these handlers?

Reply:
Mostly because it makes it so much easier to write the test suite if every field isn't present with `None` values. As long as the client and server share models, it doesn't matter, and this way the output is shorter and more human-readable.